content/doc/developer/security/secrets.adoc

```diff
@@ -80,3 +80,12 @@ Secrets spanning multiple lines (such as certificates or SSH keys) should be mas
 It is available from Jenkins 2.171.
 Plugins with older core baselines can add a dependency on the standalone library `io.jenkins.temp.jelly:multiline-secrets-ui:1.0` (or newer) with the same form element.
 See https://github.com/jenkinsci/lib-multiline-secrets-ui/blob/master/README.md[the documentation in its GitHub repository] for usage instructions and sample code.
+
+### Encryption of Secrets and Credentials
+
+Jenkins uses a master key which is a 128-bit AES key used for encrypting and decrypting secrets and credentials.
+This is stored _unencrypted_ on the Jenkins controller filesystem in `$JENKINS_HOME/secrets/master.key`, and it is used to protect the individual AES keys used for protecting secrets (commonly referred to as a "key encryption key" or "KEK").
+Jenkins uses AES in cipher block chaining (CBC) mode with PKCS#5 padding to encrypt instances of jenkinsdoc:CryptoConfidentialKey[] which are stored in `$JENKINS_HOME/secrets/` with a filename corresponding to their `CryptoConfidentialKey` id.
+Most typical usage in Jenkins and plugins are via the jenkinsdoc:Secret[] API which regenerates a fresh IV for each `Secret` encryption operation.
+Common key ids used in Jenkins include `hudson.util.Secret` (generic secrets), `com.cloudbees.plugins.credentials.SecretBytes.KEY` (some types of credentials), `jenkins.model.Jenkins.crumbSalt` (seed value used with https://www.jenkins.io/doc/book/managing/security/#cross-site-request-forgery[CSRF tokens]), and `org.jenkinsci.main.modules.instance_identity.InstanceIdentity.KEY` (identifies the Jenkins instance).
+Note that due to the presence of the unprotected master key in the Jenkins controller filesystem, this encryption does not protect against an attacker with read access to the controller filesystem.
```

Comment:
This note is a bit of duplication to the earlier mention that the key is stored unencrypted, repeated again here at the bottom. I think it would read more clearly if these mentions were mentions were combined and clarified. Also probably broken out into another paragraph.